packages/nimble-components/src/rich-text-viewer/testing/rich-text-viewer.pageobject.ts

```diff
@@ -0,0 +1,23 @@
+import type { RichTextViewer } from '..';
+
+/**
+ * Page object for the `nimble-rich-text-viewer` component.
+ */
+export class RichTextViewerPageObject {
+    public constructor(
+        private readonly richTextViewerElement: RichTextViewer
+    ) {}
+
+    public getTagName(element: Element | null | undefined): string {
```

Comment:
What's the element being passed in to these functions? Can they use the element that was passed to the constructor instead?

The goal of the page object is to abstract away details about how the component is constructed so that the tests don't need to change if that structure changes. If tests need to pass in child elements then I don't think the page object is achieving that goal.

Reply:
@vivinkrishna-ni A refactored page object to address this comment would look like this (cleanup any syntax errors) : 

```
public getViewerTagName(): string {
        return this.getViewerElement()?.tagName || '';
    }

    public getViewerTextContent(element: Element | null | undefined): string {
        return this.getViewerElement()?.textContent || '';
    }

    public getViewerElement(): Element | null | undefined {
        return this.richTextViewerElement.shadowRoot?.querySelector('#viewer')
            ?.firstElementChild;
    }
```

Reply:
`getTagName()` and `getTextContent()` is to get the particular element tag name and text content respectively, not just the first element details in the viewer container. So I have removed these functions from the page object and got the string directly in the tests.

Reply:
I thought about this some more after taking a closer look at the tests. I think it's OK for the page object to contain helper methods like you had before: we want the page object to be the thing that knows about the shadow DOM structure.

I think most of the tests would be able to depend fully on the page object if it exposed methods like
`getViewerChildTags(): string[]`
`getFirstChildTagContents(): string`
`getLastChildTagContents(): string`
`getChildTagContents(index: number): string`

Some of the more advanced tests might need to get a reference to the `viewer` element and traverse its children using DOM methods but I think that can be the exception.

Reply:
Thanks for correcting us here. As suggested, I have updated these function names and functions to get tag names and text contents.